### PR TITLE
Adds the Ice Hermit!

### PR DESCRIPTION
**Pull request by FloranOtten**
## About The Pull Request
Adds the Ice Hermit! A grumpy old man with a rifle, a shack and a distrust for outsiders.

## Why It's Good For The Game

Blood for the Blood God! Skulls for the Skull Throne! Content for the Ice Moon!

I have been entertaining the idea of surviving with barebones gear on icemoon for a while now. This should facilitate that, while also adding neat content.

## Changelog
:cl:
add: Nanotrasen Scouting forces have recently reported being shot at by an old man in a shack. Watch out for any strange log cabins!
/:cl:

### CODE REVIEW

**Comment by FloranOtten** [comment]
I promise the next one will be something besides a copy of a lavaland ghost role spawner

**Comment by wesoda25** [comment]
I’m being nit-picky so take this with a grain of salt, but some of this stuff seems odd to me. The use of fairy grass , the barricade between the trees, and the sink in the middle of the room. 

Other than that its p cool

**Comment by Kaffe-work** [comment]
could you add some pictures?

**Comment by Doosteroni** [comment]
Those gerd darn kids with their "kinetic accelimators." Get off my lawn!

File: _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm [hunk 1]
```diff
@@ -0,0 +1,452 @@
...
+"hv" = (
+/obj/structure/bonfire/prelit,
+/turf/open/floor/wood,
+/area/ruin/powered/shuttle)
+"hN" = (
+/turf/open/floor/wood,
+/area/ruin/powered/shuttle)
+"ii" = (
+/obj/effect/mob_spawn/human/hermit{
+	flavour_text = "You've been living on the ice moon for at least 40 years now! You've survived the ice whelps, hunted those damn bears and tamed a few wolves. What do those outsiders want?";
```
Comment (ATH1909) [inline _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm:39]: if the hermit's supposed to have tamed a few wolves, can you add a few friendly wolves to his shack?

File: _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm [hunk 2]
```diff
@@ -0,0 +1,452 @@
...
+"oJ" = (
+/obj/structure/sink,
+/turf/open/floor/plating,
+/area/ruin/powered/shuttle)
+"pI" = (
+/obj/item/chair/wood/wings,
+/turf/open/floor/wood,
+/area/ruin/powered/shuttle)
+"sC" = (
+/obj/item/gun/ballistic/rifle/boltaction,
```
Comment (ATH1909) [inline _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm:62]: does the hermit get any ammo for this thing?

**Comment by FloranOtten** [comment]
> I’m being nit-picky so take this with a grain of salt, but some of this stuff seems odd to me. The use of fairy grass , the barricade between the trees, and the sink in the middle of the room.

The sink serves to provide water for the botany. The fairygrass is there because I wanted to create an environment to justify not instantly freezing when you spawn, and the barricade is there purely as a bit of ramshackle fencing.

File: _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm (hunk 2 above)
Comment (FloranOtten) [reply _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm:62]: It spawns with a full clip. Any more ammo and I fear it'd just become a role that serves purely to gun down miners.

File: _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm (hunk 1 above)
Comment (FloranOtten) [reply _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm:39]: I'm not entirely sure we have any tame wolf subtypes.

Comment (Fikou) [reply _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm:39]: just make one its easy

File: _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm [hunk 3]
```diff
@@ -0,0 +1,452 @@
...
+	},
+/obj/structure/fans/tiny/invisible,
+/turf/open/floor/wood,
+/area/ruin/powered/shuttle)
+"Oe" = (
+/obj/item/seeds/plump,
+/obj/item/seeds/plump,
+/obj/item/seeds/tower,
+/obj/item/seeds/reishi,
+/obj/item/reagent_containers/food/snacks/grown/mushroom/glowshroom/glowcap,
```
Comment (Shadark) [inline _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm:108]: Maybe glowshrooms instead? Or is it intended as a way to recharge things?

Comment (FloranOtten) [reply _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm:108]: Oop, that's a mistake. I'll fix that.

File: _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm (hunk 1 above)
Comment (FloranOtten) [reply _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm:39]: I don't really know how, and learning to honour a tiny bit of fluff text seems a bit much.

Comment (ATH1909) [reply _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm:39]:
```
/mob/living/simple_animal/hostile/asteroid/wolf/tame
	name = "Tamed Wolf"
	faction = list("neutral")
```
bam, done

hell, you don't even have to make a new subtype for it, you could just override variables like you've done here with the hermit's flavour_text

**Comment by ATH1909** [comment]
note that glowshrooms decay now...

**Comment by FloranOtten** [comment]
The glowshroom cluster that's already planted really only serves as decoration, due to there being some seeds and a pregrown glowshroom there.

**Comment by Pumpkinoe** [comment]
It's pretty nice, but, a bonfire on wooden floors? That seems like a disaster waiting to happen. Also, fairygrass is kinda odd and I hope there's a tiny fan underneath the wooden airlock else it's going to be fun losing a bit of the livable atmosphere inside of your house each time you leave.

**Comment by wesoda25** [comment]
> The sink serves to provide water for the botany. The fairygrass is there because I wanted to create an environment to justify not instantly freezing when you spawn (and blue grass looks better in snow than green), and the barricade is there purely as a bit of ramshackle fencing.

Uh I understand why the sink is there, just choosing to put it in the middle of the room as opposed to against a wall seems a bit odd. I get the color scheme for fairy grass, but doesn’t it have some sort of advantage over normal grass? If not then ignore that. 

> I don't see the reason for this, this ghost role sucks even on Lavaland. At best, its a second life, at worst, its boring as hell.

I really disagree. This is a great ghost role due to its simplicity - it has near unlimited potential and creates a chance for interesting player interactions. That being said, the implementation is a bit lazy since its just copy pasting the lavaland one with minor changes. Something more icemoon themed, like a cave/igloo, would be neat.

**Comment by Pumpkinoe** [comment]
i may be keen to make an igloo sometime if you need it

**Comment by FloranOtten** [comment]
I'm not aware of any different properties fairygrass has. It just looks nice.

There is indeed a tiny fan under the doorway (though it is invisible).

**Comment by FloranOtten** [comment]
>That being said, the implementation is a bit lazy since its just copy pasting the lavaland one with minor changes. Something more icemoon themed, like a cave/igloo, would be neat.

I'm quite happy with this, but if you want to change it feel free to open up a new PR/PR on my branch. I just want to go surviving on icemoon, I don't care what the actual structure is.

**Comment by Doosteroni** [comment]
If we’re gonna have a mosin could they get the Russian battle coat? Maybe add more to the backstory with the hermit being Russian?

**Comment by Pumpkinoe** [comment]
Fairygrass glows and that's about it. It doesn't make much sense for the hermit either - What, is he actually a weed addict? Is that why he has make-you-high-grass floors?

**Comment by Pumpkinoe** [comment]
Also also, what's powering the lights that are positioned on the uh... Rocks? Won't those be left floating if someone breaks the rocks there?

**Comment by IndieanaJones** [comment]
> I really disagree. This is a great ghost role due to its simplicity - it has near unlimited potential and creates a chance for interesting player interactions. That being said, the implementation is a bit lazy since its just copy pasting the lavaland one with minor changes. Something more icemoon themed, like a cave/igloo, would be neat.

It'd be way better if it was a yeti instead of a person.

File: _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm (hunk 1 above)
Comment (Rohesie) [reply _maps/RandomRuins/IceRuins/icemoon_underground_hermit.dmm:39]: Add these var edits to the object definition. Poke on the coder channel if you want to know how to.

**Comment by Rohesie** [comment]
Closing due to staleness.
Do the requested changes and see if a maintainer shows interest in the feature if you plan to re-open it.

**Comment by Rohesie** [comment]
Request done.
I'll leave this open for 48 hours to see if any maintainers have anything else to add, or if they want to veto it.
Else, poke me and I'll merge it.

**Pull request merged**